File: android-native/app/src/main/java/flexr/social/app/ui/auth/AuthTabs.kt
package flexr.social.app.ui.auth

import androidx.compose.animation.animateColorAsState
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import flexr.social.app.core.designsystem.theme.FlexrTheme

enum class AuthTab { LOGIN, REGISTER }

/** Umschalter zwischen Anmeldung und Registrierung (`.auth-tabs` im Web). */
@Composable
fun AuthTabs(
    selected: AuthTab,
    onSelect: (AuthTab) -> Unit,
    modifier: Modifier = Modifier,
) {
    val colors = FlexrTheme.colors
    Row(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(Color.White.copy(alpha = 0.03f))
            .padding(4.dp),
    ) {
        AuthTab.entries.forEach { tab ->
            val isSelected = tab == selected
            val background by animateColorAsState(
                if (isSelected) colors.surface3 else Color.Transparent,
                label = "tabBackground",
            )
            val contentColor by animateColorAsState(
                if (isSelected) colors.plate else colors.chalkDim,
                label = "tabContent",
            )
            Box(
                modifier = Modifier
                    .weight(1f)
                    .clip(RoundedCornerShape(9.dp))
                    .background(background)
                    .clickable(enabled = !isSelected) { onSelect(tab) }
                    .padding(vertical = 11.dp),
                contentAlignment = Alignment.Center,
            ) {
                Text(
                    text = if (tab == AuthTab.LOGIN) "Einloggen" else "Registrieren",
                    style = MaterialTheme.typography.labelLarge,
                    color = contentColor,
                )
            }
        }
    }
}

File: ios/FLEXR/UI/Auth/AuthTabs.swift
import SwiftUI

enum AuthTab: String, CaseIterable, Identifiable {
    case login, register

    var id: String { rawValue }

    var title: String {
        switch self {
        case .login: "Einloggen"
        case .register: "Registrieren"
        }
    }
}

/// Umschalter zwischen Anmeldung und Registrierung (`.auth-tabs` im Web).
struct AuthTabs: View {

    let selected: AuthTab
    let onSelect: (AuthTab) -> Void

    var body: some View {
        HStack(spacing: 0) {
            ForEach(AuthTab.allCases) { tab in
                let isSelected = tab == selected
                Button { if !isSelected { onSelect(tab) } } label: {
                    Text(tab.title)
                        .flexrText(.labelLarge)
                        .foregroundStyle(isSelected ? FlexrColor.plate : FlexrColor.chalkDim)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 11)
                        .background(
                            RoundedRectangle(cornerRadius: 9, style: .continuous)
                                .fill(isSelected ? FlexrColor.surface3 : .clear)
                        )
                        .contentShape(Rectangle())
                }
                .buttonStyle(.plain)
            }
        }
        .padding(4)
        .background(
            RoundedRectangle(cornerRadius: FlexrRadius.button, style: .continuous)
                .fill(Color.white.opacity(0.03))
        )
        .animation(.easeOut(duration: 0.18), value: selected)
    }
}
